### DISCUSSION TITLE
is there a pin on the EVOfw3 that outputs a ttl level version of the serial data that is output to the USB port.

### DISCUSSION THREAD

**Discussion by BillComben** [ghoti57/evofw3#43] [Q&A]
Hi, I wish to use my EVOFW3 board powered from a USB wall power supply (not connected to a PC or laptop) but the serial monitored data needs to be input to an external ESP32 dev module using ttl levels. No serial data needs to be sent  to the EVOFW3, its just used for monitoring the Evohome messages.

Does the existing EVOFW3 board and software make this signal available on any of its accessible pins.

Thanks for your help.

**Comment by ghoti57** [comment]
Which version of the device did you have? If it's not one from indalo-tech what processor is on the board?

**Comment by BillComben** [comment]
Hi Peter, thanks for your reply, its the Indalo-tech board SSM-D2 original version. Have just seen that there is now a ESP dev board with a second USB port, that might suit my needs in the long term but for now I need to use TTL levels.

Thanks, Bill

**Comment by ghoti57** [comment]
Sorry, there's no way to get at a TTL version of the message data on that device.

Meanwhile you might like to know that I am currently working on a port to an esp device that will include MQTT as well as the usb interface  This is specifically with the idea of being able to plug the device into an isolated power socket and get the data to a host system.

**Comment by BillComben** [comment]
OK, thanks, an ESP32 version talking to an MQQT broker sounds like a good way to go, I would be interested in buying one when its available. I have just ordered a 2 USB port ESP32 dev board that I will experiment with in the short term.

In my case the EVOFW3 board is located next to an ESP32 module that acts as a hub for a network of over 20 home built ESP32 sensor boards that are installed around my property. Every minute the hub uploads all the collected data to my website and saves them in a database, they are then displayed as part of my custom home monitoring project, see http://thame.billcomben.com/homeview.php.

**Comment by ghoti57** [comment]
Bill, 

The ESP32 device is now available. 

<https://indalo-tech.onlineweb.shop/product/ramses-esp>

Regards
Peter

⁣Get BlueMail for Android ​

**Comment by BillComben** [comment]
Hi peter, thanks for your email, I have been away and only just found it.

 

Does the new ESP32 version allow me to access the data serially at the TTL level (by soldering if needed), or can the board act as a USB host, I have so far been unable to get my ESP32 S3 dual USB port boards to connect when acting as a USB host.

 

My main requirement is for board to board communications without using a MQQT broker, at present I am having to link my EVOfw3 to my esp32 via a small python program on a pc.

 

I will probably purchase one of your new boards to play with anyway.

 

Thanks, bill

**Comment by ghoti57** [comment]
Hi Bill,

I see two options if you don't want to use MQTT, both requiring firmware mods.

1. Redirect the console output to one of the unused ESP32 UARTs and picking up the TTL

2. Replace the MQTT code with something that uses a different protocol to communicate with other devices. 

Pete

⁣Get BlueMail for Android ​

**Comment by BillComben** [comment]
Thanks peter, I would consider either of your suggested options. I will consider MQQT as a short term solution till I have more free time to make the mods

 

Can you please let me know when the new version is available for purchase again. I am ready to order.

 

Thanks, Bill

**Comment by BillComben** [comment]
Hi again Peter, In preparation for using your new ESP32 board I have been reading the WIKI for the new board, I have a question for you –

 

The property that I am monitoring has 2 independent heating systems that I need to monitor, both communicate  using the Honeywell wireless protocol, one is an Evohome system and the other is a Lightwave RF system. Neither of these systems uses an 18: gateway device, both use 01: type hubs. I see that over MQQT the devices are addressed by a gateway address, how do I overcome not having any Gateway addresses. I don’t need separate sub topics, just a single topic that includes all messages monitored, I can decode the correct system from the device addresses contained  within the messages.

 

Thanks, Bill

**Comment by BillComben** [comment]
Just read your wiki again, am I correct in assuming that the gateway address refers to a unique ID of your ESP32 and not an Evohome 18:gateway device address.

 

From: Bill Comben ***@***.***> 
Sent: 30 June 2024 09:46
To: 'ghoti57/evofw3' ***@***.***>
Subject: Question on the new ESP32 wireless monitor board.

 

Hi again Peter, In preparation for using your new ESP32 board I have been reading the WIKI for the new board, I have a question for you –

 

The property that I am monitoring has 2 independent heating systems that I need to monitor, both communicate  using the Honeywell wireless protocol, one is an Evohome system and the other is a Lightwave RF system. Neither of these systems uses an 18: gateway device, both use 01: type hubs. I see that over MQQT the devices are addressed by a gateway address, how do I overcome not having any Gateway addresses. I don’t need separate sub topics, just a single topic that includes all messages monitored, I can decode the correct system from the device addresses contained  within the messages.

 

Thanks, Bill

**Comment by ghoti57** [comment]
My device IS a gateway.  If you don't tx any messages and your two 
systems are unaware of the device then no messages will contain the 
gateway 18:xxx address.
 
The only place it will be used is to register in the MQTT broker.
 
The gateway has no/needs no knowledge of the structure of your system.  
It will just blindly publish all messages to the same MQTT topic:
 
RAMSES/GATEWAY/18:xxxxxx/rx
 
All your host code has to do is subscribe to this topic and it will 
receive all the messages in a JSON format
 
{ "ts&quot;:&quot;timestamp&quot; , &quot;msg&quot;:&quot;message&quot; 
}
 
You don't even need to know the gateway address if you subscribe to
 
RAMSES/GATEWAY/+/tx
 
Peter
 
Get BlueMail for Android <https://bluemail.me>

On 30 Jun 2024, at 10:46, BillComben ***@***.*** 
***@***.***> > wrote:
 
Hi again Peter, In preparation for using your new ESP32 board I have 
been reading the WIKI for the new board, I have a question for you – 

The property that I am monitoring has 2 independent heating systems that 
I need to monitor, both communicate using the Honeywell wireless 
protocol, one is an Evohome system and the other is a Lightwave RF 
system. Neither of these systems uses an 18: gateway device, both use 
01: type hubs. I see that over MQQT the devices are addressed by a 
gateway address, how do I overcome not having any Gateway addresses. I 
don’t need separate sub topics, just a single topic that includes all 
messages monitored, I can decode the correct system from the device 
addresses contained within the messages. 

Thanks, Bill

**Comment by ghoti57** [comment]
Request a stock update on the store.

 

It will be after July 11 when I release the stock again.

 

Peter

**Comment by BillComben** [comment]
Thanks Peter, will do 

Bill
